Sources/StytchCore/StytchClientCommon/Models/StytchError.swift

```diff
@@ -1,118 +1,131 @@
-        errorMessage: "There are no biometric registrations available. Must authenticate with other methods and add a new biometric registration before calling this method."
-    )
-    static let oauthCredentialInvalid: Self = .init(
-        errorType: "oauth_apple_credential_invalid",
-        errorMessage: "The Sign In With Apple authorization credential was an invalid type/format."
-    )
-    static let oauthCredentialMissingIdToken: Self = .init(
-        errorType: "oauth_apple_missing_id_token",
-        errorMessage: "The Sign In With Apple authorization credential was missing an id_token."
-    )
-    static let oauthInvalidStartUrl: Self = .init(
-        errorType: "oauth_generic_invalid_start_url",
-        errorMessage: "The start url was invalid or improperly formatted."
-    )
-    static let oauthInvalidRedirectScheme: Self = .init(
-        errorType: "oauth_generic_invalid_redirect_scheme",
-        errorMessage: "The scheme from the given redirect urls was invalid. Possible reasons include: nil scheme, non-custom scheme (using http or https), or differing schemes for login/signup urls."
-    )
-    static let oauthASWebAuthMissingUrl: Self = .init(
-        errorType: "oauth_generic_aswebauth_missing_url",
-        errorMessage: "The underlying ASWebAuthenticationSession failed to return a URL"
-    )
+/// Error class representing when the Stytch SDK cannot reach the Stytch API.
+public class StytchAPIUnreachableError: StytchAPIError {}
 
-    static let passkeysInvalidPublicKeyCredentialType: Self = .init(
-        errorType: "passkeys_invalid_credential_type",
-        errorMessage: "The public key credential type was not of the expected type."
-    )
-    static let passkeysMissingAttestationObject: Self = .init(
-        errorType: "passkeys_missing_attestation_object",
-        errorMessage: "The public key credential is missing the attestation object."
-    )
-    static let jsonDataNotConvertibleToString: Self = .init(
-        errorType: "json_data_not_convertible_to_string",
-        errorMessage: "JSON data unable to be converted to String type."
-    )
-    static let passkeysNotSupported: Self = .init(
-        errorType: "passkeys_not_supported",
-        errorMessage: "Passkeys are unsupported on this device"
+/// Error class representing a schema error within the Stytch API.
+public class StytchAPISchemaError: StytchAPIError {}
+
+/// Error class representing an error within the Stytch SDK.
+public class StytchSDKError: StytchError {}
+
+/// Error class representing invalid input within the Stytch SDK.
+public class StytchSDKUsageError: StytchSDKError {}
+
+public extension StytchSDKError {
+    static let clientNotConfigured = StytchSDKError(
+        name: "client_not_configured",
+        description: "StytchClient not yet configured. Must include a `StytchConfiguration.plist` in your main bundle or call `StytchClient.configure(publicToken:hostUrl:)` prior to other StytchClient calls.",
```

Comment:
We probably want to genericize this since it would be shared between StytchClient and StytchB2BClient

Reply:
And I see this in the other PR, too. Nice.